compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrVisitor.kt

```diff
@@ -251,19 +251,29 @@ class Fir2IrVisitor(
     }
 
     override fun visitVarargArgumentsExpression(varargArgumentsExpression: FirVarargArgumentsExpression, data: Any?): IrElement {
-        val irReturnType = varargArgumentsExpression.typeRef.toIrType()
-        return IrVarargImpl(UNDEFINED_OFFSET, UNDEFINED_OFFSET, irReturnType,
-                            varargArgumentsExpression.varargElementType.toIrType(),
-                            varargArgumentsExpression.arguments.map { arg ->
-                                convertToIrExpression(arg).run {
-                                    if (arg is FirSpreadArgumentExpression || arg is FirNamedArgumentExpression && arg.isSpread) {
-                                        IrSpreadElementImpl(UNDEFINED_OFFSET, UNDEFINED_OFFSET, this)
-                                    }
-                                    else this
-                                }
-                            })
+        return varargArgumentsExpression.convertWithOffsets { startOffset, endOffset ->
+            IrVarargImpl(
+                startOffset,
+                endOffset,
+                varargArgumentsExpression.typeRef.toIrType(),
+                varargArgumentsExpression.varargElementType.toIrType(),
+                varargArgumentsExpression.arguments.map { arg ->
+                    convertToIrExpression(arg).run {
+                        convertToIrSpreadElementIfNeeded(arg)
+                    }
+                })
+        }
     }
 
+    private fun IrExpression.convertToIrSpreadElementIfNeeded(arg: FirExpression): IrVarargElement =
```

Comment:
Actually, it might be nicer to have a function `convertToIrVarargElement(arg: FirExpression)` which calls `convertToIrExpression` internally. That is assuming that there is no point to passing anything other than the translation of `arg` as the extension receiver.

Reply:
Indeed! Thanks for the suggestion!